clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2196,4 +2196,52 @@ def ZeroInitConstOp : CIR_Op<"llvmir.zeroinit", [Pure]>,
   let hasVerifier = 0;
 }
 
-#endif // MLIR_CIR_DIALECT_CIR_OPS
+def AsmATT : I32EnumAttrCase<"AD_ATT", 0>;
+def AsmIntel : I32EnumAttrCase<"AD_Intel", 1>;
+
+def AsmATTOrIntel : I32EnumAttr<
+  "AsmDialect",
+  "ATT or Intel",
+  [AsmATT, AsmIntel]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def CIR_InlineAsmOp : CIR_Op<"inline_asm"> {
+  let description = [{
+    The InlineAsmOp mirrors the underlying LLVM semantics with a notable
+    exception: the embedded `asm_string` is not allowed to define or reference
```

Comment:
Thanks for the documentation, this should be enforced by the verifier. Since this is still introducing the operation, let's make it even more simple: remove the comment until we add operands/results. Instead write:
```
The `cir.asm` operation represents C/C++ asm inline.
<example from testcase here>
```
Then as you add the other input operands, you improve the docs.